### PR TITLE
Model backend

### PR DESCRIPTION
**Pull request by Crash1212**

### CODE REVIEW

File: modules/product/product.resolver.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ const productsQuery = {
       };
     }
   },
+  getModelsByCategory: (parent, args) => productsService.getModelsByCategory(args.id)
```
Comment (VolodymyrTrach) [inline modules/product/product.resolver.js:25]: throw an error if model not found

File: typeDefs.js [hunk 2]
```diff
@@ -63,6 +63,10 @@ const typeDefs = gql`
     source: String
     tokenPass: String
   }
+  type Price {
+    currency: String
+    value: Int
+  }
```
Comment (VolodymyrTrach) [inline typeDefs.js:69]: remove it. we don`t need this type. We use CurrencySet for prices

**Review by VolodymyrTrach — CHANGES_REQUESTED** [review]

File: modules/product/product.resolver.js [hunk 3]
```diff
@@ -22,7 +22,18 @@ const productsQuery = {
       };
     }
   },
-  getModelsbyCategory: (parent, args) => productsService.getModelsByCategory(args.id)
+  getModelsByCategory: async (parent, args) => {
+      const models = await productsService.getModelsByCategory(args.id)
+      if (models) {
+        return models;
+      }
+      return {
```
Comment (Alan18081) [inline modules/product/product.resolver.js:30]: It should not return an error if no models are found, we just need to return empty array

**Review by Alan18081 — COMMENTED** [review]
Please, add the tests for models

File: modules/product/product.service.js [hunk 4]
```diff
@@ -111,19 +121,41 @@ class ProductsService {
     return new Product(data).save();
   }
 
+  async addModel(data) {
```
Comment (Alan18081) [inline modules/product/product.service.js:124]: I think it is better to create for the model separate folder, resolver and service

File: __tests__/model/model.query.test.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            category{
+              name {
+                value
+              }
+            },
+            name {
+              value
+            }
+            images {
+              large
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply __tests__/model/model.query.test.js:64]: fixed

File: __tests__/model/model.query.test.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            }
+          }
+        }`,
+        variables: {
+          category: categoryId
+        }
+      ,
+    }).catch(er => er );
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply __tests__/model/model.query.test.js]: fixed

File: __tests__/model/model.query.test.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    const models = res.data.getModelsByCategory;
+    
+    expect(models).toBeDefined();
+    expect(models.length).toBeGreaterThan(0);
+    expect(models[0].name).toBeInstanceOf(Array);
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply __tests__/model/model.query.test.js:80]: fixed

File: __tests__/model/model.query.test.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+              _id
+            }
+          }
+        }
+      `,
+      variables: { model: { ...newModel, category: categoryId } },
+    });
+    modelId = createModel.data.addModel._id;
+  });
+  test('Should receive all models by category id', async () => {
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply __tests__/model/model.query.test.js:43]: fixed

File: modules/product/product.service.js [hunk 9]
```diff
@@ -17,13 +19,15 @@ class ProductsService {
   }
 
   getModelsByCategory(id) {
-    return Model.find({ category: id})
+    if(!ObjectId.isValid(id))
```
[the remark this answers is not in the document]
Comment (Crash1212) [reply modules/product/product.service.js]: fixed

File: modules/product/product.service.js (hunk 4 above)
Comment (Crash1212) [reply modules/product/product.service.js]: fixed

**Pull request merged**